### PR TITLE
Fix: Relax the validation on term (when) field on create a course form

### PR DESCRIPTION
**Pull request by Formasitchijoh**
## What this PR does
This PR addresses  issue #5837 where the "Term" field in the Wiki Education Dashboard's course creation form. Previously, the validation for this field was overly strict. This update relaxes the validation criteria
## Screenshots
Before:

https://github.com/user-attachments/assets/cc7c0157-3f74-4c01-9cd9-67ce361f6d4a

After:

https://github.com/user-attachments/assets/ab169d8e-8253-4aa0-af2e-d33a145ae8e8

### CODE REVIEW

**Comment by ragesoss** [comment]
It looks like this change would modify the validation behavior of both Programs & Events Dashboard and Wiki Education Dashboard? We still want to validate the presence of a `term` in wiki_education mode.

**Comment by Formasitchijoh** [comment]
@ragesoss i have checked and the term field is only available for the Wiki Education Dashboard, not in the Programs and Events Dashboard form, please clarify me if i am wrong

**Comment by Formasitchijoh** [comment]
@ragesoss i have updated this PR, sorry i miss understood the task, so I realized the field you were refering to in this case is the when field when copying  a course from an external source in the P&E dashboard, 

I have relaxed the validation by removing the required on the input field, i retained the slug validation to ensure if a user decided to put in a value it should be valid. and the validation of the Wiki education dashboard is maintained

please i would appreciate you feedback

File: app/assets/javascripts/components/overview/course_cloned_modal.jsx [hunk 1]
```diff
@@ -339,7 +339,6 @@ const CourseClonedModal = createReactClass({
           onChange={this.updateCourse}
           value={this.state.course.term}
           value_key="term"
-          required={true}
```
Comment (ragesoss) [inline app/assets/javascripts/components/overview/course_cloned_modal.jsx:342]: We do still want to require it for `wiki_education` mode.

Comment (Formasitchijoh) [reply app/assets/javascripts/components/overview/course_cloned_modal.jsx:342]: Alright i have updated that

**Pull request merged**